app/views/reports/index.html.erb

```diff
@@ -38,62 +38,66 @@
+                            <td><%= GeneticMaterial.find(batch.geneticMaterial).name %></td>
+                            <td><%= batch.entry_control.organization.name %></td>
+                        <%	end %>
+                        <%	 if params[:purchased] != nil && params[:quality] == "" %>
+                            <%	if batch.buy == 1 %>
+                                <td><%= batch.sackAmount %></td>
+                                <td><%= number_with_precision(batch.weight, precision: 2) %> lb</td>
+                                <td><%= batch.enterCode %></td>
+                                <td><%= CocoaType.find(batch.cocoaType).name %></td>
+                                <td><%= GeneticMaterial.find(batch.geneticMaterial).name %></td>
+                                <%	 if params[:searchbox] == "" %>
+                                  <td><%= batch.entry_control.organization.name %></td>
+                                <%	 end %>
+                            <%  end %>
+                        <%	end %>
+                        <%	 if params[:purchased] == nil && params[:quality] != "" %>
+                            <%	 if batch.quality_control != nil %>
+                                <%	 if batch.quality_control.revision != nil %>
+                                    <%	 if batch.quality_control.revision.status == params[:quality].to_i %>
+                                      <td><%= batch.sackAmount %></td>
+                                      <td><%= number_with_precision(batch.weight, precision: 2) %> lb</td>
+                                      <td><%= batch.enterCode %></td>
+                                      <td><%= CocoaType.find(batch.cocoaType).name %></td>
+                                      <td><%= GeneticMaterial.find(batch.geneticMaterial).name %></td>
+                                      <%	 if params[:searchbox] == "" %>
+                                        <td><%= batch.entry_control.organization.name %></td>
+                                      <%	 end %>
+                                    <%	 end %>
+                                <%	 end %>
+                            <%	 end %>
+                        <%	end %>
+                        <%	 if params[:purchased] != nil && params[:quality] != "" %>
+                            <%	 if batch.quality_control != nil %>
+                                <%	 if batch.quality_control.revision != nil %>
+                                    <%	 if batch.quality_control.revision.status == params[:quality].to_i %>
+                                        <%	if batch.buy == 1 %>
+                                          <td><%= batch.sackAmount %></td>
+                                          <td><%= number_with_precision(batch.weight, precision: 2) %> lb</td>
+                                          <td><%= batch.enterCode %></td>
+                                          <td><%= CocoaType.find(batch.cocoaType).name %></td>
```

Comment:
Not showing type selected by quality manager

Reply:
If purchase not working correctly, use
```if params[:purchased]```
and batch is purchased if purchase decision is 1